daemon/src/model.rs

```diff
@@ -600,6 +602,70 @@ impl Timestamp {
     }
 }
 
+/// Funding rate per SETTLEMENT_INTERVAL
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub struct FundingRate(Decimal);
+
+impl FundingRate {
+    pub fn new(rate: Decimal) -> Result<Self> {
+        anyhow::ensure!(
+            rate.is_sign_positive(),
+            "Negative funding rates (ie. paid by the maker) are not supported yet"
+        );
+        anyhow::ensure!(
+            rate <= Decimal::ONE,
+            "Funding rate can't be higher than 100%"
+        );
+
+        Ok(Self(rate))
+    }
+
+    pub fn as_dec(&self) -> Decimal {
+        self.0
+    }
+}
+
+/// Fee paid on every contract renewal (a fraction of it if rolling over)
+#[derive(Debug, Clone)]
+pub struct FundingFee(u64);
+
+impl FundingFee {
+    pub fn as_u64(&self) -> u64 {
+        self.0
+    }
+}
+
+impl Default for FundingFee {
+    fn default() -> Self {
+        Self::new(Amount::ZERO, FundingRate(Decimal::ZERO), 1)
+            .expect("hard-coded values to be valid")
+    }
+}
+
+impl FundingFee {
+    pub fn new(margin: Amount, funding_rate: FundingRate, hours_to_charge: i64) -> Result<Self> {
+        anyhow::ensure!(hours_to_charge >= 1, "Can't charge for less than one hour");
+        anyhow::ensure!(
+            hours_to_charge <= SETTLEMENT_INTERVAL.whole_hours(),
+            "Can't change for more than a full settlement interval"
+        );
+        let fraction = if hours_to_charge as i64 == SETTLEMENT_INTERVAL.whole_hours() {
+            Decimal::ONE
+        } else {
+            Decimal::from(hours_to_charge)
+                .checked_div(Decimal::from(SETTLEMENT_INTERVAL.whole_hours()))
+                .context("can't establish a fraction")?
+        };
+
+        Ok(Self(
+            (Decimal::from(margin.as_sat()) * funding_rate.as_dec() * fraction)
+                .round_dp_with_strategy(0, rust_decimal::RoundingStrategy::AwayFromZero)
+                .to_u64()
+                .context("Failed to represent as u64")?,
+        ))
```

Comment:
Nit: I find it a bit weird that the math of the funding fee is in here. I think I would have expected the ctor to only check invariants and have a separate free function that does the math and constructs `FundingFee` as the last step.